### ISSUE TITLE
Unexpected behavior on 'Align' rule with statements

### ISSUE THREAD

**Issue by zheeeng** [palantir/tslint#3201]
### Bug Report

- __TSLint version__: 5.7.0
- __TypeScript version__: Version 2.5.2
- __Running TSLint via__:  VSCode

#### TypeScript code being linted

```ts
let foo = 2
let bar = 3
; [foo, bar] = [bar, foo]
```

with `tslint.json` configuration:

```json
"align": [true, "parameters", "statements" ],
```

#### Actual behavior

tip: `[tslint] statements are not aligned (align)`

#### Expected behavior

no warning

**Comment by zheeeng** [comment]
seems caused by leading semicolon, it requires a valid form like 
```
let foo = 2
let bar = 3;
[foo, bar] = [bar, foo]
```

**Comment by samal84** [comment]
I would expect to get this error in this case.

If you prefix the third line with "; " then the statement in the third line indeed does not align with the statement on the line before, as the third line statement starts two characters more indented.

**Comment by JoshuaKGoldberg** [comment]
Since `;`s are sometimes inserted by ASI compensation techniques _(sigh, ASI)_, it seems reasonable that the `align` rule could take in an option to allow this.

**Comment by JoshuaKGoldberg** [comment]
### 💀 _**It's time!**_ 💀

**TSLint is being deprecated** and no longer accepting pull requests for major new changes or features. See #4534. 😱

If you'd like to see this change implemented, you have two choices:
* **Recommended:** Check if this is available in [ESLint](https://eslint.org) + [typescript-eslint](https://typescript-eslint.io) ✅
* _Not Recommended: Fork TSLint locally_ 🤷‍♂️

👋 It was a pleasure open sourcing with you!

_If you believe this message was posted here in error, please comment so we can re-open the issue!_

**Comment by JoshuaKGoldberg** [comment]
🤖 Beep boop! 👉  **TSLint is deprecated** 👈 _(#4534)_  and you should **switch to [typescript-eslint](https://typescript-eslint.io)**! 🤖 

🔒 This issue is being locked to prevent further unnecessary discussions. Thank you! 👋
